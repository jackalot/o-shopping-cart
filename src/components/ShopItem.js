import React, { useState } from "react";
import '../css-component-files/ShopItem.css'
function ShopItem (props) {
    const [quantity, setQuantity] = useState(0);
    function changeQuantity(newQuantity) {
        setQuantity(newQuantity);
    }
    return <div className="ShopItem-Container">
        <div className="ShopItem-ItemName">
       {props.itemName}
        </div>
        <div className="ShopItem-Price">
        Price: {props.itemPrice}
        </div>
        <label onChange={changeQuantity} for={props.itemName + "quantity"}>Quantity</label>
        <input className="QuantityInput" name={props.itemName + "quantity"} type="number"></input>
        <button className="AddToCartBtn" onClick={()=> props.handler(props.itemName, props.itemPrice)}>Add To Cart!</button>
    </div>
}
export default ShopItem;