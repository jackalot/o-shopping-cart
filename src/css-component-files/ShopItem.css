.ShopItem-Container {
    background-color: #FDB750;
    width: fit-content;
    display: flex;
    flex-direction: column;
    justify-content: center;
}
.ShopItem-ItemName {
    font-size: 40px;
    padding-left: 20px;
    padding-right: 20px;
}
.ShopItem-Price {
    font-size: 20px;
    padding-left: 20px;
    padding-right: 20px;
}
.QuantityInput {
    padding-left: 20px;
    padding-right: 20px;
}
.AddToCartBtn {
    border-color: #010100;
}