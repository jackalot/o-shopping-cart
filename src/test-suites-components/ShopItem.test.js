import React from "react";
import "@testing-library/jest-dom"
import userEvent from "@testing-library/user-event"; 
import { getByDisplayValue, getByLabelText, getByRole, render } from "@testing-library/react";
import ShopItem from "../components/ShopItem";

describe("ShopItem component", () => {
    it("renders the Add To Cart button", ()=> {
        const { getByRole } = render(<ShopItem></ShopItem>);
        expect(getByRole("button").className).toMatch("AddToCartBtn");
    })
    it("has a item name of Taco and a price of 10.00", ()=> {
        const { container } = render(<ShopItem></ShopItem>)
        expect(container).toMatchSnapshot();
    })
    it("ShopItem has a quantity field", () => {
        const { container } = render(<ShopItem itemName="Taco" itemPrice="10.00" />);
        const quantity = getByLabelText(container, "Quantity");
        expect(quantity.textContent).toBe('');
    })
})