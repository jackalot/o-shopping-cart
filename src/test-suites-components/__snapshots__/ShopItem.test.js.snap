// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopItem component has a item name of Taco and a price of 10.00 1`] = `
<div>
  <div
    class="ShopItem-Container"
  >
    <div
      class="ShopItem-ItemName"
    />
    <div
      class="ShopItem-Price"
    >
      Price: 
    </div>
    <label
      for="undefinedquantity"
    >
      Quantity
    </label>
    <input
      class="QuantityInput"
      id="undefinedquantity"
      name="undefinedquantity"
      type="number"
    />
    <button
      class="AddToCartBtn"
    >
      Add To Cart!
    </button>
  </div>
</div>
`;
